src/query/expression/src/types/decimal.rs

```diff
@@ -378,7 +379,34 @@ impl Decimal for i128 {
     }
 
     fn from_float(value: f64) -> Self {
-        value.to_i128().unwrap()
+        // still needs to be optimized.
```

Comment:
It's great if we have some unit test for the from_float bound check, for example:
```

    #[test]
    fn test_from_float_positive() {
        let value = 123.456;
        let result = from_float(value);
        assert_eq!(result, 123); // The fractional part should be truncated
    }

    #[test]
    fn test_from_float_negative() {
        let value = -123.456;
        let result = from_float(value);
        assert_eq!(result, -123); // The fractional part should be truncated
    }

    #[test]
    fn test_from_float_zero() {
        let value = 0.0;
        let result = from_float(value);
        assert_eq!(result, 0); // Zero should be converted to zero
    }

    #[test]
    fn test_from_float_large() {
        let value = 1e20;
        let result = from_float(value);
        assert_eq!(result, i128::MAX); // Large values should be capped to i128::MAX
    }

    #[test]
    fn test_from_float_small() {
        let value = -1e20;
        let result = from_float(value);
        assert_eq!(result, i128::MIN); // Small values should be capped to i128::MIN
    }

    #[test]
    fn test_from_float_sub_one_positive() {
        let value = 0.123;
        let result = from_float(value);
        assert_eq!(result, 0); // Values less than one should be converted to zero
    }

    #[test]
    fn test_from_float_sub_one_negative() {
        let value = -0.123;
        let result = from_float(value);
        assert_eq!(result, 0); // Values less than negative one should be converted to zero
    }

```

Reply:
let me add